Retry upon connection error

tzuhsial opened the pull request:
Modifications I added to process 5000 documents for 4~5 hours

PS: I believe you will need to add python2 compatibility before packaging it to pypi....

tzuhsial committed:
Retry upon connection error

tzuhsial commented:
@yuwen41200 Please check

yuwen41200 commented on ckipclient/ckipclient.py:
```diff
@@ -37,9 +37,20 @@ def __init__(self, ip: str, port: int, usr: str, pwd: str, safe: bool = True):
+            try:
+                results = self._segment(text, pos)
+                status = True
+            except ConnectionError:
```
How about checking other errors as well?
For example:
```python
def safe_segment(message):
    while True:
        try:
            message = segment(message) if message else None
        except ElementTree.ParseError as err:
            print('ElementTree.ParseError:', err, flush=True)
            print('wait 10 seconds...', flush=True)
            time.sleep(10)
        except ConnectionError as err:
            print('ConnectionError:', err, flush=True)
            print('wait 10 seconds...', flush=True)
            time.sleep(10)
        except OSError as err:
            print('OSError:', err, flush=True)
            print('wait 10 seconds...', flush=True)
            time.sleep(10)
        else:
            return message
```

tzuhsial replied:
Changed function name to 'safe_segment'
Since the parsing code is yours, I'll leave the rest of the errors to you.

yuwen41200 commented on ckipclient/ckipclient.py:
```diff
@@ -37,9 +37,20 @@ def __init__(self, ip: str, port: int, usr: str, pwd: str, safe: bool = True):
-        self.counter = 0
 
-    def segment(self, text: str, pos: bool = True) -> list:
+    def segment(self, text: str, pos: bool = True):
```
For consistency, please keep the type hints.

tzuhsial replied:
Oops...accidentally modified that part

yuwen41200 reviewed (commented):
Thanks! Please see my comments.

yuwen41200 commented:
@iammrhelo Regarding Python 2 compatibility, sorry but I have no plan to add Python 2 support.

tzuhsial committed:
add retry count to 'safe_segment'

tzuhsial commented:
@yuwen41200 
If it doesn't support Python2, 
I think you should add that into the Readme.(and not package it to pypi)

Anyways, for this commit
1. added missing type hint
2. changed function name for consistency
   - 'segment' -> 'safe_segment'
   - '_segment' -> 'segment'
3. add 'retry' parameter in 'safe_segment' to specify number of retries
Please check, thanks!

tzuhsial commented:
@yuwen41200 Forgot to mention, thanks a lot for your work!

yuwen41200 reviewed (approved):
Thanks.

yuwen41200 commented:
@iammrhelo 
OK, I added a notice in README.
But... I should not upload it to PyPI?! Why?
As far as I know, supporting Python 2 is NOT a must.
I've already specified its Python version correctly, conforming to the requirements as stated in:
https://packaging.python.org/tutorials/distributing-packages/
